packages/langium/src/lsp/language-server.ts

```diff
@@ -39,7 +40,10 @@ export function startLanguageServer(services: LangiumSharedServices): void {
                 hoverProvider: {},
                 renameProvider: {
                     prepareProvider: true
-                }
+                },
+                semanticTokensProvider: languages.some(e => e.lsp.SemanticTokenProvider !== undefined)
+                    ? DefaultSemanticTokenOptions
```

Comment:
A good idea, but I don't really see a use case for that right now. This would be a bit annoying to override, since it would either belong in the shared services (which seems kind of weird), or we simply take the first options that are appear in all services, which also seems wrong.

I would wait until someone requests this feature.